Bottom: 03e0d17919c572f6b56a2e64ede76820e556d4c2
Top:    21fc1256c673952b3d563855361504be260009ac
Author: Ben Greear <greearb@candelatech.com>
Date:   2014-07-05 09:06:38 -0700

ath10k: Disable beacon-miss firmware logic.

Not sure it works all that well in multi-vdev
environments anyway, since it only look at 2 vdevs?

Needs some testing, and maybe further improvements to
do useful beacon-loss detection with multiple vdevs.

Signed-off-by: Ben Greear <greearb@candelatech.com>


---

diff --git a/drivers/net/wireless/ath/ath10k/wmi.c b/drivers/net/wireless/ath/ath10k/wmi.c
index 00167b4..907e715 100644
--- a/drivers/net/wireless/ath/ath10k/wmi.c
+++ b/drivers/net/wireless/ath/ath10k/wmi.c
@@ -5597,6 +5597,11 @@ static struct sk_buff *ath10k_wmi_10_1_op_gen_init(struct ath10k *ar)
 		config.roam_offload_max_vdev = 0; /* disable roaming */
 		config.roam_offload_max_ap_profiles = 0; /* disable roaming */
 		config.num_peer_keys = __cpu_to_le32(TARGET_10X_NUM_PEER_KEYS_CT);
+
+		/* Disable beacon-miss logic, old code had it at 2 vdevs, which is not
+		 * nearly enough for us anyway..
+		 */
+		config.bmiss_offload_max_vdev = 0;
 	} else {
 		skid_limit = TARGET_10X_AST_SKID_LIMIT;
 		config.roam_offload_max_vdev =
@@ -5605,6 +5610,9 @@ static struct sk_buff *ath10k_wmi_10_1_op_gen_init(struct ath10k *ar)
 		config.roam_offload_max_ap_profiles =
 			__cpu_to_le32(TARGET_10X_ROAM_OFFLOAD_MAX_AP_PROFILES);
 		config.num_peer_keys = __cpu_to_le32(TARGET_10X_NUM_PEER_KEYS);
+		config.bmiss_offload_max_vdev =
+			__cpu_to_le32(TARGET_10X_BMISS_OFFLOAD_MAX_VDEV);
+
 	}
 	config.num_msdu_desc = __cpu_to_le32(ar->htt.max_num_pending_tx);
 	config.ast_skid_limit = __cpu_to_le32(skid_limit);
@@ -5622,9 +5630,6 @@ static struct sk_buff *ath10k_wmi_10_1_op_gen_init(struct ath10k *ar)
 	config.scan_max_pending_reqs =
 		__cpu_to_le32(TARGET_10X_SCAN_MAX_PENDING_REQS);
 
-	config.bmiss_offload_max_vdev =
-		__cpu_to_le32(TARGET_10X_BMISS_OFFLOAD_MAX_VDEV);
-
 	config.num_mcast_groups = __cpu_to_le32(TARGET_10X_NUM_MCAST_GROUPS);
 	config.num_mcast_table_elems =
 		__cpu_to_le32(TARGET_10X_NUM_MCAST_TABLE_ELEMS);
